lib/nested_filter.ex

```diff
@@ -5,57 +5,45 @@ defmodule NestedFilter do
 
   @doc """
   Take a (nested) map and filter out any keys with specified values in the
-  filter_values list.
+  values_to_reject list.
   """
-  @spec drop_by_value(map :: map(), filter_values :: list()) :: map()
-  def drop_by_value(map, filter_values) do
-    cond do
-      is_nested_map?(map) ->
-        new_map = map
-                  |> Enum.reduce(%{}, fn({key, val}, acc) ->
-                     Map.put(acc, key, drop_by_value(val, filter_values)) end)
-        Map.drop(new_map, filterable_keys(new_map, filter_values))
-      is_map(map) ->
-        Map.drop(map, filterable_keys(map, filter_values))
-      true ->
-        map
-    end
+  @spec drop_by_value(map :: map(), values_to_reject :: list()) :: map()
+  def drop_by_value(map, values_to_reject) when is_map(map) do
+    map
+    |> Enum.reduce(%{}, fn
+      ({key, val}, acc) ->
+        cleaned_val = drop_by_value(val, values_to_reject)
+        if cleaned_val in values_to_reject do
+          acc
+        else
+          Map.put(acc, key, cleaned_val)
+        end
+    end)
+  end
+
+  def drop_by_value(not_map, _) do
```

Comment:
You'll see that I made several function heads for the `drop_by_value` and `drop_by_key` functions. This eliminates the map check in your `cond` statement.